e2e/tests/workspace/cdr-version-upgrade.spec.ts

```diff
@@ -1,69 +1,62 @@
 
-describe('Workspace CDR Version upgrade', () => {
+describe('Workspace CDR Version Upgrade modal', () => {
     beforeEach(async () => {
         await signIn(page);
     });
 
-   test('Clicking the CDR version upgrade flag pops up the upgrade modal', async () => {
-        const workspaceCard: WorkspaceCard = await createWorkspace(page, config.altCdrVersionName);
-        await workspaceCard.clickWorkspaceName();
+   test('Clicking Cancel and Upgrade buttons', async () => {
 
-        const workspacePage: WorkspaceBase = new WorkspaceDataPage(page);
-        expect(await workspacePage.getCdrVersion()).toBe(config.altCdrVersionName);
-        const newVersionFlag = await workspacePage.getNewCdrVersionFlag();
-        await newVersionFlag.click();
+      const workspaceCard: WorkspaceCard = await createWorkspace(page, config.altCdrVersionName);
+      const workspaceName = await workspaceCard.clickWorkspaceName();
 
-        const modal = new CdrVersionUpgradeModal(page);
-        expect(await modal.isLoaded()).toBe(true);
+      const workspacePage: WorkspaceBase = new WorkspaceDataPage(page);
+      const cdrVersion = await workspacePage.getCdrVersion();
+      expect(cdrVersion).toBe(config.altCdrVersionName);
 
-        // hit cancel
-        const cancelButton = await modal.getCancelButton();
-        await cancelButton.click();
+      let modal = await launchCdrUpgradeModal(page);
 
-        // new version flag remains
-        await workspacePage.getNewCdrVersionFlag();
+      // Clicking the Cancel
+      let cancelButton = await modal.getCancelButton();
+      await cancelButton.click();
 
-        // cleanup
-        await workspacePage.deleteWorkspace()
-    });
-
-    test('Clicking the CDR version upgrade button opens the Duplicate Workspace Page', async () => {
-        const workspaceCard: WorkspaceCard = await createWorkspace(page, config.altCdrVersionName);
-        const workspaceName = await workspaceCard.clickWorkspaceName();
+      // CDR version flag remains
+      await workspacePage.getNewCdrVersionFlag();
 
-        const workspacePage: WorkspaceBase = new WorkspaceDataPage(page);
+      // Clicking the Upgrade button opens the Duplicate Workspace Page
+      modal = await launchCdrUpgradeModal(page);
+      const upgradeButton = await modal.getUpgradeButton();
+      await upgradeButton.click();
 
-        const newVersionFlag = await workspacePage.getNewCdrVersionFlag();
-        await newVersionFlag.click();
+      const duplicationPage = new WorkspaceEditPage(page);
+      const upgradeMessage = await duplicationPage.getCdrVersionUpgradeMessage();
+      expect(upgradeMessage).toContain(workspaceName);
+      expect(upgradeMessage).toContain(`${config.altCdrVersionName} to ${config.defaultCdrVersionName}.`);
 
-        const modal = new CdrVersionUpgradeModal(page);
-        expect(await modal.isLoaded()).toBe(true);
+      cancelButton = await duplicationPage.getCancelButton();
```

Comment:
This is a different Cancel button from the button above, so I'd prefer a different variable name.  Then both of these can be const.

Reply:
Done. Thanks.